mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewPopup.java

```diff
@@ -0,0 +1,143 @@
+package org.exarhteam.iitc_mobile;
+
+import android.annotation.SuppressLint;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.util.AttributeSet;
+import android.view.WindowManager;
+import android.webkit.WebChromeClient;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+@SuppressLint("SetJavaScriptEnabled")
+public class IITC_WebViewPopup extends WebView {
+    private WebSettings mSettings;
+    private IITC_Mobile mIitc;
+    private SharedPreferences mSharedPrefs;
+    private final String mDesktopUserAgent = "Mozilla/5.0 (X11; Linux x86_64; rv:17.0)" +
+            " Gecko/20130810 Firefox/17.0 Iceweasel/17.0.8";
+    private String mMobileUserAgent;
+
+    private Dialog mDialog;
+
+
+    // init web view
+    private void iitc_init(final Context c) {
+        if (isInEditMode()) return;
+        mIitc = (IITC_Mobile) c;
+        mSettings = getSettings();
+        mSettings.setJavaScriptEnabled(true);
+        //mSettings.setSavePassword(true);
+        //mSettings.setSaveFormData(true);
+        setVerticalScrollBarEnabled(false);
+        setHorizontalScrollBarEnabled(false);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT)
+            setWebContentsDebuggingEnabled(true);
+
+        setWebChromeClient(new WebChromeClient() {
+            @Override
+            public void onCloseWindow(WebView view) {
+                Log.d("close popup window");
+                mDialog.dismiss();
+                view.destroy();
```

Comment:
is "destroy" here required?
and why is the superclass function skipped?

...BTW I'm a totally noob with android+webviews

Reply:
Not adding it leads to some webviews not beeing destroyed on my phone

I'm also a noob with those two...